django_saml2_auth/views.py

```diff
@@ -149,12 +161,11 @@ def acs(r):
 
     user_identity = authn_response.get_identity()
     if user_identity is None:
-        return HttpResponseRedirect(get_reverse([denied, 'denied', 'django_saml2_auth:denied']))
+        return HttpResponseRedirect(get_reverse(
+            [denied, 'denied', 'django_saml2_auth:denied']))
 
-    user_email = user_identity[settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {}).get('email', 'Email')][0]
-    user_name = user_identity[settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {}).get('username', 'UserName')][0]
-    user_first_name = user_identity[settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {}).get('first_name', 'FirstName')][0]
-    user_last_name = user_identity[settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {}).get('last_name', 'LastName')][0]
+    user_name = user_identity[settings.SAML2_AUTH.get(
+        'ATTRIBUTES_MAP', {}).get('username', 'UserName')][0]
```

Comment:
Hmm.. this code removed from here seems to always just use the `...[0]` for every value it reads from the `user_identity` dict. Maybe the value is always a list and so the `if isinstance(..., list)` part is unnecessary?

Reply:
All those are, but as this is generic it should check if it is list or not.

Reply:
Ah, I see. Did you check that there actually are some attributes that pysaml2 stores into user_identity and they're not lists? (Because from this it seems that it stores everything always as a list.)

Reply:
Removed the list check as we don't need it anyways.